fvm/src/syscalls/bind.rs

```diff
@@ -94,6 +94,35 @@ fn memory_and_data<'a, K: Kernel>(
     Ok((Memory::new(mem), data))
 }
 
+fn update_gas_available_snapshot<K: Kernel>(
+    caller: &mut Caller<InvocationData<K>>,
+) -> Result<(), Trap> {
+    let gas_available = caller.data().kernel.gas_available();
+    let gas_used = gas_available - caller.data().gas_available_snapshot;
+    let fuel_used = caller.data().kernel.price_list().gas_to_fuel(gas_used);
+    caller.consume_fuel(fuel_used)?;
+
+    caller.data_mut().gas_available_snapshot = gas_available;
+    Ok(())
+}
```

Comment:
Just make this a method on the invocation data, then call `consume_fuel` afterwards.

Reply:
@arajasek I think you solved this in a follow up commit, but the commit only says "implement fip-0032". Could you please loop back here?

Reply:
Yes, it was solved.